src/envs/behavior_options.py

```diff
@@ -0,0 +1,1012 @@
+}
+
+
+def get_body_ids(  # type: ignore
+    env,
+    include_self: bool = False,
+    grasping_with_right: bool = False,
+) -> List[int]:
+    """Function to return list of body_ids for all objects for collision
+    checking depending on whether navigation or grasping/placing is being
+    done."""
+    ids = []
+    for obj in env.scene.get_objects():
+        if isinstance(obj, URDFObject):
+            # We want to exclude the floor since we're always floating and
+            # will never practically collide with it, but if we include it
+            # in collision checking, we always seem to collide.
+            if obj.name != "floors":
+                ids.extend(obj.body_ids)
+
+    if include_self:
+        ids.append(env.robots[0].parts["left_hand"].get_body_id())
+        ids.append(env.robots[0].parts["body"].get_body_id())
+        ids.append(env.robots[0].parts["eye"].get_body_id())
+        if not grasping_with_right:
+            ids.append(env.robots[0].parts["right_hand"].get_body_id())
+
+    return ids
+
+
+def detect_collision(bodyA: int, object_in_hand: int = None) -> bool:
+    """Detects collisions between objects in the scene (except for the object
+    in the robot's hand)"""
+    collision = False
+    for body_id in range(p.getNumBodies()):
+        if body_id in [bodyA, object_in_hand]:
+            continue
+        closest_points = p.getClosestPoints(bodyA, body_id, distance=0.01)
+        if len(closest_points) > 0:
+            collision = True
+            break
+    return collision
+
+
+def detect_robot_collision(robot) -> bool:  # type: ignore
+    """Function to detect whether the robot is currently colliding with any
+    object in the scene."""
+    object_in_hand = robot.parts["right_hand"].object_in_hand
+    return (detect_collision(robot.parts["body"].body_id)
+            or detect_collision(robot.parts["left_hand"].body_id)
+            or detect_collision(robot.parts["right_hand"].body_id,
+                                object_in_hand))
+
+
+def get_aabb_volume(lo: np.ndarray, hi: np.ndarray) -> float:
+    """Simple utility function to compute the volume of an aabb."""
+    dimension = hi - lo
+    return dimension[0] * dimension[1] * dimension[2]
+
+
+def get_closest_point_on_aabb(xyz: List, lo: np.ndarray,\
+    hi: np.ndarray) -> List[float]:
+    """Get the closest point on an aabb from a particular xyz coordinate."""
+    closest_point_on_aabb = [0.0, 0.0, 0.0]
+    for i in range(3):
+        # if the coordinate is between the min and max of the aabb, then
+        # use that coordinate directly
+        if xyz[i] < hi[i] and xyz[i] > lo[i]:
+            closest_point_on_aabb[i] = xyz[i]
+        else:
+            if abs(xyz[i] - hi[i]) < abs(xyz[i] - lo[i]):
+                closest_point_on_aabb[i] = hi[i]
+            else:
+                closest_point_on_aabb[i] = lo[i]
+
+    return closest_point_on_aabb
+
+
+def reset_and_release_hand(env) -> None:  # type: ignore
```

Comment:
because the type of env isn't actually a `BehaviorEnv`, but rather an `igibson.envs.behavior_env.BehaviorEnv`, which may not get imported, especially on CI

Reply:
cool, also keep comments in the code describing this and the other answers you've been giving on github :)